include/miopen/miopen.h

```diff
@@ -6582,6 +6583,39 @@ MIOPEN_EXPORT miopenStatus_t miopenBackendInitialize(miopenBackendDescriptor_t d
 // CLOSEOUT BackendAPI DOXYGEN GROUP
 #endif // MIOPEN_BETA_API
 
+#ifdef MIOPEN_BETA_API
+
+// GLU APIs
+/** @addtogroup glu
+ *
+ *  @{
+ */
+
+/*! @brief Execute a GLU forward contiguous layer
+ *
+ * @param handle                   MIOpen handle (input)
+ * @param inputDesc                Tensor descriptor for data input tensor (input)
+ * @param inputSplitDesc           Tensor descriptor for splitted data input tensor (input)
+ * @param inputFirstHalf           First half input data tensor  (input)
+ * @param inputSecondHalf          Second half input data tensor (input)
+ * @param dim                      Dimension to split (input)
+ * @param outputDesc               Tensor descriptor for output data tensor (input)
+ * @param output                   Output data tensor (output)
+ * @return                         miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t miopenGLUForward(miopenHandle_t handle,
```

Comment:
GLU is basically an operator that receives only input and output tensors not inputFirstHalf and inputSecondhalf and output, receives dimention as a parameter.
And divides it in half based on dimention , then multiplies the first half of the divided input by the second half to which sigmoid is applied.
Since the work we do is creating replicas of pytorch or tensorflow, be sure to refer to the definitions of pytorch and tensorflow. (https://pytorch.org/docs/stable/generated/torch.nn.GLU.html) 
The API shared externally in miopen.h is used directly by users. Do not change the definition except in some unavoidable cases.
Therefore, in order to implement GLU in the same way as before, you must understand tensor view and slice. See https://github.com/ROCm/MIOpen/pull/2883. Please note that this PR is currently under review, so some details may change.

Reply:
I've updated the interface to accept only one input, and it operates as follows:
```
    const size_t gid = blockIdx.x * blockDim.x + threadIdx.x;
    if(gid >= N)
        return;

    size_t innerIdx       = gid % inner_size;
    size_t splittedDimIdx = ((gid - innerIdx) / inner_size) % splitedDim_size;
    size_t outerIdx =
        (gid - innerIdx - splittedDimIdx * inner_size) / (inner_size * splitedDim_size);
    size_t inputIdx1 =
        outerIdx * splitDim_size * inner_size + splittedDimIdx * inner_size + innerIdx;
    size_t inputIdx2 = outerIdx * splitDim_size * inner_size +
                       (splittedDimIdx + splitedDim_size) * inner_size + innerIdx;

    FLOAT_ACCUM val1 = CVT_FLOAT2ACCUM(input[inputIdx1]);
    FLOAT_ACCUM val2 = sigmoid(CVT_FLOAT2ACCUM(input[inputIdx2]));
    FLOAT_ACCUM val  = val1 * val2;
    output[gid]      = CVT_ACCUM2FLOAT(val);
```
For each element in the output, I locate the corresponding element in the two halves of the input, multiply the first element by sigmoid of the second. Could you please review this implementation to ensure it meets the requirements?

Reply:
If you don't understand, look up tensor contiguous, non-contiguous, slicing.

Reply:
I have changed my implementation as your comment (assuming only the first dim is spliced). Could you please check it?
```
template <typename TI, typename TO>
__device__ void
GLUFwdCOntiguousKernel(const TI* __restrict__ input, TO* __restrict__ output, long N)
{
    const TI* inputFirstHalf  = input;
    const TI* inputSecondHalf = input + N;
    const size_t gid          = blockIdx.x * blockDim.x + threadIdx.x;
    if(gid >= N)
        return;

    FLOAT_ACCUM val1 = CVT_FLOAT2ACCUM(inputFirstHalf[gid]);
    FLOAT_ACCUM val2 = sigmoid(CVT_FLOAT2ACCUM(inputSecondHalf[gid]));
    FLOAT_ACCUM val  = val1 * val2;
    output[gid]      = CVT_ACCUM2FLOAT(val);
}
```